spec.md

```diff
@@ -2070,103 +2099,20 @@ module Foo:
-  define xp = probe(p)
-  connect p, x
-  connect y, p
-```
-
-### Probes and Const Types
+## The `read`{.firrtl} Expressions
 
-Probe types may target `const`{.firrtl} signals, but cannot use `rwprobe`{.firrtl} with a constant signal to produce a `RWProbe<const T>`{.firrtl}.
+Both `Probe`{.firrtl} and `RWProbe`{.firrtl} may be read from using the `read`{.firrtl} expression.
 
-### Exporting References to Nested Declarations
+Keep in mind that probes are intended for verification and aren't intended to be synthesizable.
+A `read`{.firrtl} expression can reach arbitrarily far into the module hierarchy to read from a distant circuit component, which is arguably unphysical at worst and potentially expensive at best.
 
-Nested declarations (see [@sec:nested-declarations]) may be exported:
+## Forcing
 
-``` firrtl
-module RefProducer :
-  input a : UInt<4>
-  input en : UInt<1>
-  input clk : Clock
-  output thereg : Probe<UInt>
+A `RWProbe`{.firrtl} may be forced using the `force`{.firrtl} and `force_initial`{.firrtl} commands.
+When you force a component, it will ignore all other drivers and take on the value supplied by the force statement instead.
+The `release`{.firrtl} and `release_initial`{.firrtl} statements end the forcing, reconnecting the circuit component to its usual driver.
 
-  when en :
-    reg myreg : UInt, clk
-    connect myreg, a
-    define thereg = probe(myreg)
-```
-
-### Forwarding References Upwards
-
-Define statements can be used to forward a child module's reference further up the hierarchy:
-
-``` firrtl
-module Foo :
-  output p : Probe<UInt>
-  ; ...
-
-module Forward :
-  output p : Probe<UInt>
-
-  inst f of Foo
-  define p = f.p
-```
-
-Define statements may narrow a probe of an aggregate to a sub-element using static expression:
-
-``` firrtl
-module Foo :
-  output p : Probe<UInt[2]>[2]
-  ; ...
-
-module Forward :
-  output p : Probe<UInt>
-
-  inst f of Foo
-  define p = f.p[0][1]
-```
-
-### Forwarding References Downwards
-
-Define statements can also be used to forward references down the hierarchy using input reference-type ports, which are allowed but should be used carefully as they make it possible to express invalid reference paths.
-
-See [@sec:input-probe-references] for more details, a small example is given below:
-
-``` firrtl
-module UnusedInputRef :
-  input r : Probe<UInt<1>>
-
-module ForwardDownwards :
-  input in : UInt<1>
-
-  inst u of UnusedInputRef
-  define u.r = probe(in)
-```
-
-## Force and Release
-
-To override existing drivers for a `RWProbe`{.firrtl}, the `force`{.firrtl} statement is used, and released with `release`{.firrtl}.
-Force statements are simulation-only constructs and may not be supported by all implementations.
-They are similar to the verification statements (e.g., `assert`{.firrtl}) in this regard.
-
-These are two variants of each, enumerated below:
+Like `read`{.firrtl}, the force statements are verification constructs.
```

Comment:
Just checking, "force" is not code-formatted because it's referring to the force "family" of statements?

Reply:
Yeah.